public/themes/pterodactyl/js/frontend/files/src/actions.js

```diff
@@ -316,6 +319,76 @@ class ActionsClass {
         });
     }
 
+    addToList(event) {
+      const parent = $(event.target).closest('tr');
+
+      const nameBlock = $(parent).find('td[data-identifier="name"]');
+      const delPath = decodeURIComponent(nameBlock.data('path'));
+      const delName = decodeURIComponent(nameBlock.data('name'));
+
+      var item = delPath + delName;
+
+      //Determine if we're removing or adding the item
+      if(files.indexOf(item) != -1) {
+        files.splice($.inArray(item, files), 1)
+        parent.removeClass('warning').delay(200);
+      } else {
+        files.push(item);
+        fileselements.push(parent);
+
+        parent.addClass('warning').delay(200);
+      }
+
+    }
+
+    deleteSelected() {
+      var nameBlock;
+
+      swal({
+          type: 'warning',
+          title: '',
+          text: 'Are you sure you want to delete <code>' +
+                $.each(files, function(key, value) {
+                  value + ", ";
+                })
+                + '</code>? There is <strong>no</strong> reversing this action.',
+          html: true,
+          showCancelButton: true,
+          showConfirmButton: true,
+          closeOnConfirm: false,
+          showLoaderOnConfirm: true
+      }, () => {
+          $.ajax({
+              type: 'DELETE',
+              url: `${Pterodactyl.node.scheme}://${Pterodactyl.node.fqdn}:${Pterodactyl.node.daemonListen}/server/file/f/${files}`,
```

Comment:
I'm wondering if we shouldn't have a different endpoint for mass deletions to pass data as a JSON object, rather than a URL argument.